app/server/utils/blockchain_tasks.py

```diff
@@ -86,7 +114,11 @@ def get_token_decimals(token):
                                                       'function': 'decimals'
                                          })
 
-    return _execute_synchronous_task(decimals_sig)
+    if current_app.config['IS_TEST']:
```

Comment:
😬

yeahh very non-ideal to test this way. where is this run? unit tests, or integration tests?
if it's integration tests, it should be running its default behavior to really see the integration. if it's unit tests, you can make a mock of the worker at the point of usage to customize the return.

in ruby you can do something like: `expect(worker).to receive(:get_token_decimals).and_return(18)`
from a quick search, the closest thing i see is: https://realpython.com/python-mock-library/#managing-a-mocks-return-value
but you'd probably know better than me if there's something better

Reply:
it's for integration tests. Currently it doesn't work because the response is supposed to be returned from the celery worker, but given all the tests are actually inside docker-compose, there's probably no reason we couldn't have them run.  I'll create a task for myself